src/values.rs

```diff
@@ -511,6 +515,36 @@ impl Value {
                 Self::Null => {
                     unimplemented!("null is meant as return value for nullable for now")
                 }
+                Self::IntRange { x, y } => {
+                    if let CoreTypeConcrete::IntRange(info) = Self::resolve_type(ty, registry)? {
+                        let inner = registry.get_type(&info.ty)?;
+                        let inner_layout = inner.layout(registry)?;
+
+                        let mut data = Vec::with_capacity(2);
+
+                        let member_ptr = x.to_ptr(arena, registry, &info.ty)?;
+                        data.push((inner_layout, 0, member_ptr));
+
+                        let (struct_layout, offset) = inner_layout.extend(inner_layout)?;
+
+                        let member_ptr = y.to_ptr(arena, registry, &info.ty)?;
+                        data.push((inner_layout, offset, member_ptr));
+
+                        let ptr = arena.alloc_layout(struct_layout.pad_to_align()).as_ptr();
+
+                        for (layout, offset, member_ptr) in data {
+                            std::ptr::copy_nonoverlapping(
+                                member_ptr.cast::<u8>().as_ptr(),
+                                ptr.byte_add(offset),
+                                layout.size(),
+                            );
+                        }
```

Comment:
Why not just write it directly instead of creating a vec, pushing two items and looping over it?